File: Assignment3/Server/src/main/java/bgu/spl/net/impl/BidiMessaging/Messages/NOTIFICATIONmessage.java
package bgu.spl.net.impl.BidiMessaging.Messages;

import bgu.spl.net.impl.BidiMessaging.Message;

public class NOTIFICATIONmessage extends Message {
    char notificationType;
    String postingUser;
    String Content;

    public NOTIFICATIONmessage(char notificationType, String postingUser, String content) {
        super((short) 9);
        this.notificationType = notificationType;
        this.postingUser = postingUser;
        Content = content;
    }

    public char getNotificationType() {
        return notificationType;
    }

    public String getPostingUser() {
        return postingUser;
    }

    public String getContent() {
        return Content;
    }
}

File: Assignment3/assignment3Client/src/NotificationMessage.cpp
//
// Created by dapool on 07/01/19.
//

#include "NotificationMessage.h"

NotificationMessage::NotificationMessage(): Message(9), postingUser(nullptr), type(-1), content(""){}

NotificationMessage::NotificationMessage(std::string _postingUser, char _type, std::string _content): Message(9),
                postingUser(std::move(_postingUser)), type(_type) , content(_content) {}

NotificationMessage::NotificationMessage(const NotificationMessage &notificationMessage): Message(9),
    postingUser(notificationMessage.postingUser), type(notificationMessage.type),
    content(notificationMessage.content){}

NotificationMessage& NotificationMessage::operator=(const NotificationMessage &notificationMessage) {
    postingUser = notificationMessage.postingUser;
    type = notificationMessage.type;
    content = notificationMessage.content;
}

std::string NotificationMessage::getPostingUser() {
    return postingUser;
}

short NotificationMessage::getType() {
    return type;
}

std::string NotificationMessage::getContent() {
    return content;
}


